src/main/java/com/google/api/codegen/transformer/GapicTestTransformer.java

```diff
@@ -0,0 +1,277 @@
+  private final TestValueGenerator valueGenerator;
+
+  public GapicTestTransformer(ValueProducer valueProducer) {
+    this.valueGenerator = new TestValueGenerator(valueProducer);
+  }
+
+  public List<Interface> getGrpcInterfacesToMock(Model model, ApiConfig apiConfig) {
+    Map<String, Interface> interfaces = new LinkedHashMap<>();
+
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      if (!service.isReachable()) {
+        continue;
+      }
+      interfaces.put(service.getFullName(), service);
+      InterfaceConfig interfaceConfig = apiConfig.getInterfaceConfig(service);
+      for (MethodConfig methodConfig : interfaceConfig.getMethodConfigs()) {
+        String reroute = methodConfig.getRerouteToGrpcInterface();
+        if (!Strings.isNullOrEmpty(reroute)) {
+          Interface targetInterface = model.getSymbolTable().lookupInterface(reroute);
+          interfaces.put(reroute, targetInterface);
+        }
+      }
+    }
+
+    return new ArrayList<Interface>(interfaces.values());
+  }
+
+  public List<MockGrpcMethodView> createMockGrpcMethodViews(SurfaceTransformerContext context) {
+    List<Method> methods = context.getInterface().getMethods();
+    ArrayList<MockGrpcMethodView> mocks = new ArrayList<>(methods.size());
+    for (Method method : methods) {
+      MethodTransformerContext methodContext = context.asRequestMethodContext(method);
+      String requestTypeName =
+          methodContext.getTypeTable().getAndSaveNicknameFor(method.getInputType());
+      String responseTypeName =
+          methodContext.getTypeTable().getAndSaveNicknameFor(method.getOutputType());
+      MethodConfig methodConfig = methodContext.getMethodConfig();
+      mocks.add(
+          MockGrpcMethodView.newBuilder()
+              .name(methodContext.getNamer().getApiMethodName(method, VisibilityConfig.PUBLIC))
+              .requestTypeName(requestTypeName)
+              .responseTypeName(responseTypeName)
+              .grpcStreamingType(methodConfig.getGrpcStreamingType())
+              .streamHandleTypeName(methodContext.getNamer().getStreamingServerName(method))
+              .build());
+    }
+    return mocks;
+  }
+
+  public List<MockServiceUsageView> createMockServices(
+      SurfaceNamer namer, Model model, ApiConfig apiConfig) {
+    List<MockServiceUsageView> mockServices = new ArrayList<>();
+
+    for (Interface service : getGrpcInterfacesToMock(model, apiConfig)) {
+      MockServiceUsageView mockService =
+          MockServiceUsageView.newBuilder()
+              .className(namer.getMockServiceClassName(service))
+              .varName(namer.getMockServiceVarName(service))
+              .implName(namer.getMockGrpcServiceImplName(service))
+              .registerFunctionName(namer.getServerRegisterFunctionName(service))
+              .build();
+      mockServices.add(mockService);
+    }
+
+    return mockServices;
+  }
+
+  public InitCodeContext createRequestInitCodeContext(
```

Comment:
I think this should actually live in `InitCodeTransformer`. There's nothing really test-specific about this.